### PR TITLE
Add multiple features to PPO

### PR DESCRIPTION
**Pull request by instance01**
Add:
* Value loss clipping
* LR annealing
* KL rollback
* Epsilon annealing

Also removed some redundant whitespaces.

### CODE REVIEW

File: PPO/PPO.py [hunk 1]
```diff
@@ -160,12 +171,15 @@ def learn(self, num_learn, last_value, next_done):
       # calculate gradient
       self.optimiser.zero_grad()
       loss.backward()
-      nn.utils.clip_grad_norm_(self.model_old.parameters(), 0.5)
+      nn.utils.clip_grad_norm_(self.model.parameters(), 0.5)
```
Comment (instance01) [inline PPO/PPO.py:174]: @DarylRodrigo I need you to look at this, I hope I didn't misunderstand something.
